realtime/utils.js

```diff
@@ -393,9 +394,13 @@ function attachAspectSubject(sample) {
   let promiseArr = [];
   if (featureToggles.isFeatureEnabled('attachSubAspFromDB')) {
     const getAspectPromise = sample.aspect ? Promise.resolve(sample.aspect) :
-      aspectModel.scope({ method: ['name', aspName]}).find();
+    sequelize.query('SELECT id, name, tags FROM "Aspects" WHERE name LIKE :name and "isDeleted" = 0 LIMIT 1',
+    { replacements: { name: aspName  }, type: sequelize.QueryTypes.SELECT });
+
     const getSubjectPromise = sample.subject ? Promise.resolve(sample.subject) :
-      subjectModel.scope({ method: ['absolutePath', subName]}).find();
+    sequelize.query('SELECT id, "absolutePath", name, tags FROM "Subjects" WHERE "absolutePath" LIKE :abspath and "isDeleted" = 0 LIMIT 1',
+    { replacements: { abspath: subName  }, type: sequelize.QueryTypes.SELECT });
+
```

Comment:
@iamigo these changes seems to have fixed the memory issue. It is able to take in 1000 bulk upserts of 1200 samples every 1000ms.

Reply:
Amazing! Can we isolate just this change and see if it holds?